changelog/27289.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:improvement
```

Comment:
I marked this as improvement because I believe the json editor retaining the data is a regression from when we replaced partials with glimmerized child components [here](https://github.com/hashicorp/vault/pull/11672/files#diff-286f9a8e64268bca1328f28c04f60f42c6bf3c1bbc1082b0e26f221b8f923713R30) which made the component track the `data` whereas previously it was cleared and restored to all default values on back.

https://github.com/hashicorp/vault/blob/475f3243f9b15c84ca68029092831afcac085e53/ui/app/components/tool-actions-form.js#L50-L55

However, based on the customer feedback repeatedly wrapping the same data is the desired behavior.